File: BaseballGame.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    let s = []

    for (let i of operations){
        if (i === '+'){
            s.push(s[s.length - 1] + s[s.length - 2])
        }else if (i === 'D'){
            s.push(s[s.length - 1] * 2)
        }else if (i === 'C'){
            s.pop()
        }else{
            s.push(parseInt(i))
        }
    }

    let res = 0
    for (let i of s){
        res += i
    }
    return res

    // Time: O(n)
    // Space: O(n)
};

File: BaseballGame.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> s = new Stack<Integer>();

        for (String i : operations){
            if (i.equals("+")){
                s.push(s.peek() + s.get(s.size() - 2));
            }
            else if(i.equals("D")){
                s.push(s.peek() * 2);
            }
            else if(i.equals("C")){
                s.pop();
            }else{
                s.push(Integer.parseInt(i));
            }
        }

        int res = 0;
        for (int i : s){
            res += i;
        }
        return res;
    }

    // Time: O(n)
    // Space: O(n)
}
